src/providers/object-provider.js

```diff
@@ -176,44 +176,40 @@ export default class YamcsObjectProvider {
     async #getTelemetryDictionary() {
         if (this.dictionaryLoaded) {
             return this.dictionary;
+        } else if (this.dictionaryPromise) {
+            return this.dictionaryPromise;
         }
 
-        const dictionary = await this.#fetchTelemetryDictionary(this.url, this.instance, this.folderName);
-        this.dictionary = dictionary;
-        this.dictionaryLoaded = true;
-        this.roleStatusTelemetry.dictionaryLoadComplete();
+        this.dictionaryPromise = this.#fetchTelemetryDictionary(this.url, this.instance, this.folderName)
+            .finally(() => {
+                this.dictionaryLoaded = true;
+                this.roleStatusTelemetry.dictionaryLoadComplete();
+            });
 
-        return dictionary;
+        return this.dictionaryPromise;
```

Comment:
Can we have this always return the promise?

Reply:
This shouldn't be an issue, the way the logic is setup, it should only every be called one time. Once `this.dictionaryLoaded` is set, it's never changed.